### PR TITLE
Add BIT sol for Sleepysort

### PR DESCRIPTION
**Pull request by HarryWangATX**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-898.mdx [hunk 1]
```diff
@@ -122,3 +124,78 @@ public class Sleepy {
...
+```java
+import java.io.*;
+import java.util.*;
+
+public class Sleepy {
+	static int n;
+	static int[] a, c, ans;
+	public static void main(String[] args) throws IOException {
+		BufferedReader in = new BufferedReader(new FileReader("sleepy.in"));
+		PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("sleepy.out")));
```
Comment (SansPapyrus683) [inline solutions/usaco-898.mdx:143]: just use `new PrintWriter("sleepy.out")` and put @ bottom

File: solutions/usaco-898.mdx [hunk 2]
```diff
@@ -122,3 +124,78 @@ public class Sleepy {
...
+		n = Integer.parseInt(in.readLine());
+		a = new int[n+1];
+		c = new int[n+1];
+		ans = new int[n+1];
```
Comment (SansPapyrus683) [inline solutions/usaco-898.mdx:147]: spaces

File: solutions/usaco-898.mdx [hunk 3]
```diff
@@ -122,3 +124,78 @@ public class Sleepy {
...
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Sleepy {
+	static int n;
+	static int[] a, c, ans;
```
Comment (SansPapyrus683) [inline solutions/usaco-898.mdx:140]: better names

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-898.mdx [hunk 4]
```diff
@@ -122,3 +124,78 @@ public class Sleepy {
...
+	public static int query(int x) {
+		int ans = 0;
+		for (int i = x; i > 0; i -= lowbit(i)) {
+			ans += c[i];
+		}
+		return ans;
+	}
+}
+```
+
```
Comment (ryanchou-dev) [inline solutions/usaco-898.mdx:199]:
```suggestion
```

**Review by ryanchou-dev — APPROVED** [review]
preferably add comments, but code looks good (after kevin's suggestions)

**Review by nwatx — APPROVED** [review]

**Comment by HarryWangATX** [comment]
@SansPapyrus683 bump :)

**Comment by SansPapyrus683** [comment]
i'd ask you to put it in a class but oh well

**Pull request merged**